packages/shared/lib/core/layer-2/utils/signTransactionWithLedger.ts

```diff
@@ -0,0 +1,36 @@
+import { Ledger } from '@core/ledger'
+import { EvmTransactionData } from '../types'
+import { Platform } from '@core/app'
+import { MILLISECONDS_PER_SECOND, sleep } from '@core/utils'
+
+export async function signTransactionWithLedger(transaction: EvmTransactionData, bip32Path: string): Promise<string> {
+    await Ledger.signEvmTransaction(transaction, bip32Path)
+
+    let isSigning = true
+    let signedTransaction: string = ''
+
+    function signingFinished(_signedTransaction: string): void {
+        isSigning = false
+        signedTransaction = _signedTransaction
+    }
+
+    Platform.onEvent('evm-signed-transaction', ({ signedTransaction }) => {
+        signingFinished(signedTransaction)
+    })
+
+    const pollingInterval = 100
+    const timeoutInSeconds = 10
+    const loopIterations = (timeoutInSeconds * MILLISECONDS_PER_SECOND) / pollingInterval
+
+    for (let count = 0; count < loopIterations; count++) {
+        if (!isSigning) {
+            if (signedTransaction) {
+                return Promise.resolve(signedTransaction)
+            } else {
+                return Promise.reject('Rejected')
+            }
+        }
+        await sleep(pollingInterval)
+    }
+    return Promise.reject('Timeout')
```

Comment:
This works very nicely. I think we should replace the error messages with locale's. The error on l.30 can say something like: `You must open the Ethereum app on your Ledger device`?

The error on l.35 could use the `error.ledger.temout` locale.

Reply:
the error on line 30 will be thrown if the user `Rejects` the signing. So it should rather be `Signing was rejected by the Ledger device`.

Reply:
It's also thrown when the wrong app is open, but I'll add fixing this as a polish task. Ideally we should use Matt's function to check which app is open and show a pop-up if it's the wrong one.